src/ticketing/TicketingContext.tsx

```diff
@@ -65,6 +70,19 @@ const ticketingReducer: TicketingReducer = (
         isRefreshingFareContracts: false,
       };
     }
+    case 'UPDATE_SENT_FARE_CONTRACTS': {
+      const currentFareContractOrderIds = action.fareContracts.map(
+        (fc) => fc.orderId,
+      );
+      return {
+        ...prevState,
+        sentFareContracts: action.fareContracts,
+        reservations: prevState.reservations.filter(
+          (r) => !currentFareContractOrderIds.includes(r.orderId),
+        ),
```

Comment:
We remove reservations when we retrieve the fare contract. We may need to do that when we receive the sent fare contract as well, but I am not completely sure.

Reply:
Then we will need to do that as well, so the `FareContractAndReservationList` can work as intended, thanks for pointing it out!

Reply:
I can confirm that the reservations are filtered out when we receive the sent fare contract. 😄